Test.System.Containers.Filesystem/CreateNewImageTests.cs

```diff
@@ -0,0 +1,118 @@
+using System.Containers.Tasks;
+using System.Diagnostics;
+
+namespace Test.System.Containers.Tasks;
+
+[TestClass]
+public class CreateNewImageTests
+{
+    [TestMethod]
+    public void CreateNewImage_Baseline()
+    {
+        DirectoryInfo newProjectDir = new DirectoryInfo(Path.Combine(Path.GetTempPath(), nameof(CreateNewImage_Baseline)));
+
+        if (newProjectDir.Exists)
+        {
+            newProjectDir.Delete(recursive: true);
+        }
+
+        newProjectDir.Create();
+
+        ProcessStartInfo info = new ProcessStartInfo
+        {
+            WorkingDirectory = newProjectDir.FullName,
+            FileName = "dotnet",
+            Arguments = "new console -f net7.0"
+        };
+
+        // Create the project to pack
+        Process dotnetNew = Process.Start(info);
+        Assert.IsNotNull(dotnetNew);
+        dotnetNew.WaitForExit();
+        Assert.AreEqual(0, dotnetNew.ExitCode);
+
+        info.Arguments = "build --configuration release";
+
+        Process dotnetPublish = Process.Start(info);
+        Assert.IsNotNull(dotnetPublish);
+        dotnetPublish.WaitForExit();
+        Assert.AreEqual(0, dotnetPublish.ExitCode);
+
+        CreateNewImage task = new CreateNewImage();
+        task.BaseRegistry = "https://mcr.microsoft.com";
+        task.BaseImageName = "dotnet/runtime";
+        task.BaseImageTag = "6.0";
+
+        task.OutputRegistry = "http://localhost:5010";
+        task.PublishDirectory = Path.Combine(newProjectDir.FullName, "bin", "release", "net7.0");
+        task.ImageName = "dotnet/testimage";
+        task.WorkingDirectory = "app/";
+        task.Entrypoint = "dotnet build";
+        task.EntrypointArgs = "";
+
+        Assert.IsTrue(task.Execute());
+        newProjectDir.Delete(true);
+    }
+
+    [TestMethod]
+    public void ParseContainerProperties_EndToEnd()
+    {
+        DirectoryInfo newProjectDir = new DirectoryInfo(Path.Combine(Path.GetTempPath(), nameof(ParseContainerProperties_EndToEnd)));
+
+        if (newProjectDir.Exists)
+        {
+            newProjectDir.Delete(recursive: true);
+        }
+
+        newProjectDir.Create();
+
+        ProcessStartInfo info = new ProcessStartInfo
+        {
+            WorkingDirectory = newProjectDir.FullName,
+            FileName = "dotnet",
+            Arguments = "new console -f net7.0",
+            RedirectStandardOutput = true,
+            RedirectStandardError = true,
+        };
+
+        Process dotnetNew = Process.Start(info);
+        Assert.IsNotNull(dotnetNew);
+        dotnetNew.WaitForExit();
+        Assert.AreEqual(0, dotnetNew.ExitCode);
+
+        info.Arguments = "build --configuration release";
+
+        Process dotnetPublish = Process.Start(info);
+        Assert.IsNotNull(dotnetPublish);
+        dotnetPublish.WaitForExit();
+        Assert.AreEqual(0, dotnetPublish.ExitCode);
+
+        ParseContainerProperties pcp = new ParseContainerProperties();
```

Comment:
What's the goal of this test? Since you were creating a new project I thought it was "does it work end-to-end" but you're explicitly calling the tasks so not really. For `ParseContainerProperties` in particular you shouldn't need any filesystem stuff, right?

Reply:
This is E2E via tasks. Taking the outputs of ParseContainerProperties and plugging them directly into `CreateNewImage`. So the filesystem stuff is for calling `CreateNewImage` directly.